### PR TITLE
Updated Pocl and PyOpenCL versions

### PR DESCRIPTION
**Pull request by astro-friedel**
Also separated the requirements.txt for each project

### CODE REVIEW

File: mirge/requirements-docker.txt
Comment (inducer) [inline mirge/requirements-docker.txt:1]: Is this meaningfully different than the `requirements.txt` file that's already in `mirge`?

File: Dockerfile [hunk 1]
```diff
@@ -3,11 +3,12 @@ FROM quay.io/jupyter/scipy-notebook:lab-4.0.9
...
 USER root
-RUN apt-get update
-RUN apt-get -y install libpocl-dev python3-numpy python3-pyopencl
+RUN mamba install -y -p /opt/conda -c conda-forge numpy pocl pyopencl islpy pip
 USER 1000
-COPY requirements.txt .
-RUN pip install -r requirements.txt
+COPY mirge/requirements-docker.txt m-requirements.txt
+RUN pip install -r m-requirements.txt
+COPY Parsl_tutorial/requirements.txt p-requirements.txt
```
Comment (inducer) [inline Dockerfile]: Would it make sense to rename this from `Parsl_tutorial` to just `parsl`?

File: Dockerfile [hunk 2]
```diff
@@ -3,11 +3,12 @@ FROM quay.io/jupyter/scipy-notebook:lab-4.0.9
...
 
 USER root
-RUN apt-get update
-RUN apt-get -y install libpocl-dev python3-numpy python3-pyopencl
+RUN mamba install -y -p /opt/conda -c conda-forge numpy pocl pyopencl islpy pip
 USER 1000
-COPY requirements.txt .
-RUN pip install -r requirements.txt
+COPY mirge/requirements-docker.txt m-requirements.txt
+RUN pip install -r m-requirements.txt
```
Comment (inducer) [inline Dockerfile]: Are you sure that this `pip` is from the environment that `mamba` installed into above?

Comment (astro-friedel) [reply Dockerfile]: Yes, there is only 1 pip in the image and it is in coda

File: mirge/requirements-docker.txt
Comment (astro-friedel) [reply mirge/requirements-docker.txt:1]: Yes, the branches that are referenced in the requirements.txt
loopy.git@ncsa-west-fixes
arraycontext.git@more-ary-creation
are not visible to me when I try to create the image. Once these are visible we can collapse that requirements.txt

File: Dockerfile (hunk 1 above)
Comment (astro-friedel) [reply Dockerfile]: I should be able to do that.

**Pull request closed without merging**